src/whylogs/app/logger.py

```diff
@@ -216,20 +278,47 @@ def log(
             return None
 
         if self.should_rotate():
-            self.rotate_time()
+            self._rotate_time()
 
+        
+        
+        #segmnet check  in case segments are just keys
+        self.profile_full_dataset=profile_full_dataset
+        if (segments is not None):
+            self.set_segments(segments)
+        
+               
+    
         if features is None and feature_name is None:
-            return
+            return 
 
         if features is not None and feature_name is not None:
             raise ValueError("Cannot specify both features and feature_name")
 
         if features is not None:
-            self._profiles[-1].track(features)
+            # full profile
+            if self.full_profile_check():
+                self._profiles[-1]["full_profile"].track(features)
+            
+            if self.segments:
+                self.log_segments(pd.DataFrame(features))
         else:
-            self._profiles[-1].track_datum(feature_name, value)
+            if (self.segments is None ) or ((self.segments is not None) and self.profile_full_dataset):
+                self._profiles[-1]["full_profile"].track_datum(feature_name, value)
+            
+            if self.segments:
```

Comment:
I'm not sure I understand the following code block in terms of how the data is segmented? Given it's a piece of core logic, it's worth extracting to a method with documentation

Reply:
Yeah this is a bit confusing, since it is a tracking a segmented datum. I think they way i did is inconsistent with the groupby method, since that method keeps the segmented feature in the data frame. I will fix this.